### ISSUE TITLE
Generated types ClientErrorStatus & ServerErrorStatus TS2514: A tuple type cannot be indexed with a negative value.

### ISSUE THREAD

**Issue by somethingnew2-0** [fabien0102/openapi-codegen#136]
Generated code throws a typescript error from https://github.com/fabien0102/openapi-codegen/blob/main/plugins/typescript/src/templates/utils.ts#L14-L15

```
ERROR in src/api/apiUtils.ts:9:21
TS2514: A tuple type cannot be indexed with a negative value.
     7 |
     8 | export type ClientErrorStatus = Exclude<
  >  9 |   ComputeRange<500>[-1],
       |                     ^^
    10 |   ComputeRange<400>[-1]
    11 | >;
    12 | export type ServerErrorStatus = Exclude<


ERROR in src/api/apiUtils.ts:10:21
TS2514: A tuple type cannot be indexed with a negative value.
     8 | export type ClientErrorStatus = Exclude<
     9 |   ComputeRange<500>[-1],
  > 10 |   ComputeRange<400>[-1]
       |                     ^^
    11 | >;
    12 | export type ServerErrorStatus = Exclude<
    13 |   ComputeRange<600>[-1],

ERROR in src/api/apiUtils.ts:13:21
TS2514: A tuple type cannot be indexed with a negative value.
    11 | >;
    12 | export type ServerErrorStatus = Exclude<
  > 13 |   ComputeRange<600>[-1],
       |                     ^^
    14 |   ComputeRange<500>[-1]
    15 | >;
    16 |

ERROR in src/api/apiUtils.ts:14:21
TS2514: A tuple type cannot be indexed with a negative value.
    12 | export type ServerErrorStatus = Exclude<
    13 |   ComputeRange<600>[-1],
  > 14 |   ComputeRange<500>[-1]
       |                     ^^
    15 | >;
    16 |
```

**Comment by kallebornemark** [comment]
Also facing this. Currently solving it by manually removing the `[-1]` from the generated files.

**Comment by fabien0102** [comment]
I guess something changes in TypesScript 😅 

![image](https://user-images.githubusercontent.com/1761469/220296169-2dd445da-bd8f-4f21-bdfe-2eae71b861e3.png)
![image](https://user-images.githubusercontent.com/1761469/220296209-b90ada9b-6c2d-4a11-9663-573070abd9e4.png)

**Comment by fabien0102** [comment]
Fixed in `6.1.1`! 

Please note that the expected type is a union of all error status number like this:
<img width="516" alt="image" src="https://user-images.githubusercontent.com/1761469/220310805-d6cab79e-ccf4-4090-a0ea-443e6226ebbb.png">

So you need to use `number` instead of `-1` (not supported anymore).

This little trick is helpful if you want to discriminate on error types later.

```ts
type ComputeRange<
    N extends number,
    Result extends Array<unknown> = []
> = Result["length"] extends N
    ? Result
    : ComputeRange<N, [...Result, Result["length"]]>;
type ClientErrorStatus = Exclude<ComputeRange<500>[number], ComputeRange<400>[number]>;
type ServerError = Exclude<ComputeRange<600>[number], ComputeRange<500>[number]>;


// This is generated by openapi-codegen if you have a specific 422 response defined
type APIError = {
    status: Exclude<ClientErrorStatus, 422>,
    payload: {
        message: string;
    }
} | {
    status: 422,
    payload: {
        message: string;
        field: string;
    }
}

function handleError(err: APIError) {
    if (err.status === 422) {
        console.log(err.payload.field) // I can access the specific property for 422
        return;
    }

    err.status // possible values: 400 to 499 without the handled 422
}
```
